### PR TITLE
Address the issue of non-Android targets not undergoing linting

### PR DESCRIPTION
**Pull request by mohammadkahelghi-grabtaxi**
add lint targets for kotlin libraries

### CODE REVIEW

File: rules/kotlin/kotlin.bzl [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+
+def kt_jvm_library_interal(
+        name,
+        exec_properties = None,
+        lint_options = {},
+        **attrs):
+    lint_sources_target = "_" + name + "_lint_sources"
+    lint_baseline = lint_options.get("baseline", None)
+
+    lint_sources(
```
Comment (arunkumar9t2) [inline rules/kotlin/kotlin.bzl:17]: Can we add an `if` check on `lint_options.enabled` to add these `lint_sources` and `lint` only when it is true? 

This way we don't by default enable android lint on kotlin targets. Later we can generate this option from grazel.

File: rules/kotlin/kotlin.bzl [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+        name = lint_sources_target,
+        srcs = attrs.get("srcs"),
+        baseline = lint_baseline,
+        lint_config = lint_options.get("lint_config", None),
+    )
+
+    kt_jvm_library(
+        name = name,
+        srcs = attrs.get("srcs"),
```
Comment (arunkumar9t2) [inline rules/kotlin/kotlin.bzl:26]: Consider enabling lint only when `srcs` is not empty.

File: rules/kotlin/kotlin.bzl [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+        tags = attrs.get("tags", default = []) + [LINT_ENABLED],
```
Comment (arunkumar9t2) [inline rules/kotlin/kotlin.bzl:27]: Can we use `**attrs`?

Similar to https://github.com/grab/grab-bazel-common/blob/7c9717930a9c2cbfd270324a68001fc1227155a8/rules/kotlin/kotlin.bzl#L27

File: tools/aapt_lite/src/main/java/com/google/devtools/build/android/BUILD.bazel [hunk 4]
```diff
@@ -18,7 +18,7 @@ java_library(
     ],
 )
 
-kotlin_library(
+kt_jvm_library(
```
Comment (arunkumar9t2) [inline tools/aapt_lite/src/main/java/com/google/devtools/build/android/BUILD.bazel:21]: Can we just keep `kotlin_library` as the public symbol? Updating the impl in `rules/kotlin/kotlin.bzl` will allow us to use `kotlin_library` for both bazel-common and consuming projects.

**Review by arunkumar9t2 — CHANGES_REQUESTED** [review]

File: rules/kotlin/kotlin.bzl (hunk 1 above)
Comment (mohammadkahelghi-grabtaxi) [reply rules/kotlin/kotlin.bzl]: done

File: rules/kotlin/kotlin.bzl (hunk 3 above)
Comment (mohammadkahelghi-grabtaxi) [reply rules/kotlin/kotlin.bzl]: we will get duplicated arg because we need to modify tags and deps

**Review by arunsampathkumar-grabtaxi — APPROVED** [review]

**Pull request merged**